packages/runtime/container-runtime/src/blobManager.ts

```diff
@@ -83,6 +85,11 @@ export class BlobManager {
             async () => this.getStorage().readBlob(response.id),
         );
 
+        if (this.runtime.attachState !== AttachState.Attached) {
```

Comment:
That's almost never is the right check. I believe you want to check for this.runtime.attachState === AttachState.Detached.
Handling Attaching case would require more thinking and more code, so I think for now we need to add assert that it is not supported (I assume so)